src/Sarif.Multitool/MergeCommand.cs

```diff
@@ -44,7 +51,7 @@ public static int Run(MergeOptions mergeOptions)
 
                 Directory.CreateDirectory(mergeOptions.OutputFolderPath);
 
-                FileHelpers.WriteSarifFile(combinedLog, outputName, formatting);
+                FileHelpers.WriteSarifFile(_fileSystem, combinedLog, outputName, formatting);
```

Comment:
>WriteSarifFile [](start = 28, length = 14)

Oh, yeah, this reinforces my idea for `FileHelpers`. Having seen the usage pattern, I _do_ think you should do this:

1. Rename `FileHelpers` to `FileHelper` (both the class and the source file).
2. Make `FileHelper` an instance class, and pass `IFileSystem` to its ctor.
3. In the command class ctors, instantiate a `FileHelper` object and pass it the `IFileSystem`. Probably the command classes don't need to retain their own copy of the `IFileSystem` (certainly this particular class doesn't).

Reply:
I am not going to take your suggestion at this time. The reason is that I am not sure that file helpers should survive at all. it is less helpful now that we've eliminated the need for the contract resolver. it is redundant with some other utility code and extension methods. Also, I'd like to see how the Convert command can/should be changed to improve file access to enable testing. all of this is more work than I can invest. :) I made some improvements of testability and am drawing the line here. I will look further into improving the situation next time the hood is up.

---
In reply to: [229509097](https://github.com/Microsoft/sarif-sdk/pull/1125#discussion_r229509097) [](ancestors = 229509097,229508810)